test/src/test_string.cpp

```diff
@@ -205,6 +205,29 @@ BOOST_AUTO_TEST_CASE(json_codec_string_should_decode_escaped_unicode) {
   BOOST_CHECK_EQUAL(string_parse("\"\\u20AC\""), "\xE2\x82\xAC");
 }
 
+BOOST_AUTO_TEST_CASE(json_codec_string_should_decode_surrogate_pairs) {
+  BOOST_CHECK_EQUAL(string_parse("\"I \\ud83d\\udc95 Unicode\""), "I \xf0\x9f\x92\x95 Unicode");
```

Comment:
Would be a nice with an ASCII comment here, to show what the Unicode string is:
`"I [TWO HEARTS] Unicode"`